src/AstGen.zig

```diff
@@ -1718,6 +1718,62 @@ fn structInitExpr(
         }
     }
 
+    // Temporary Arena to simplify freeing the HashMaps
+    var dup_arena = std.heap.ArenaAllocator.init(astgen.arena);
+    const dup_alloc = dup_arena.allocator();
```

Comment:
Please don't make an arena-backed arena -- this is a quite weird structure. Back your allocations directly by the AstGen arena, and if the freeing is difficult, just don't bother freeing -- the AstGen arena only lives while AstGen is running, which is brief.